From: HackerOS Linux System <hackeros068@gmail.com>
Subject: [PATCH 23/35] hackeros: rozszerzony ptrace i memory forensics dla red team

Dodaje sysctl hackeros_ptrace_redteam_scope dla kontrolowanego bypass
YAMA w dedykowanym red team session (volatility3, gdb, pwndbg)
z pelnym audit trail.

---
 kernel/ptrace.c | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/kernel/ptrace.c
+++ b/kernel/ptrace.c
@@ -1598,3 +1598,11 @@
 int hackeros_ptrace_scope_default = 1;
 EXPORT_SYMBOL(hackeros_ptrace_scope_default);
 #endif
+
+#ifdef CONFIG_HACKEROS_REDTEAM
+/* HackerOS Red Team: ptrace forensics parametry.
+ * hackeros_ptrace_redteam_scope: gdy=1, procesy z grupy hackeros-redteam
+ * pomijaja YAMA ptrace_scope. Kazda operacja auditowana. (domyslnie off) */
+static __maybe_unused int hackeros_ptrace_redteam_scope __read_mostly = 0;
+static const int hackeros_ptrace_redteam_present __initconst = 1;
+#endif /* CONFIG_HACKEROS_REDTEAM */
--
2.43.0
